### PR TITLE
No deal notice

### PR DESCRIPTION
**Pull request by vanitabarrett**
Trello: https://trello.com/c/FLS2G6uj/370-frontend-display-no-deal-content-notice-banner-on-relevant-pages

## What

- Adds a header-notice component. This could eventually be rolled into our existing [notice component](https://components.publishing.service.gov.uk/component-guide/notice), but at the moment the use case is only within government-frontend and could be iterated multiple times as it's still being tested, so it's easier keeping it within one app.
- Adds the concept of a No Deal Notice to specific content items to match the [Whitehall flag](https://github.com/alphagov/whitehall/pull/5249).

## Why
We want to flag no-deal content to users as it may potentially not be relevant to them at the moment.

Screenshot 2020-01-24 at 12 06 23

### CODE REVIEW

File: app/presenters/content_item/no_deal_notice.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            "module": "track-click",
+            "track-category": "no_deal_notice",
+            "track-action": link["href"],
+            "track-label": link["title"],
+          },
+        }
+      end
+
+      no_deal_links
+    end
```
Comment (laurentqro) [inline app/presenters/content_item/no_deal_notice.rb]: I think you could get away with just:

```
no_deal_notice.map do |link|
        {
          title: link["title"],
          href: link["href"],
          data_attributes: {
            "module": "track-click",
            "track-category": "no_deal_notice",
            "track-action": link["href"],
            "track-label": link["title"],
          },
        }
      end
```

File: app/views/content_items/case_study.html.erb [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 <%= render 'shared/publisher_metadata_with_logo' %>
 <%= render 'govuk_publishing_components/components/notice', @content_item.withdrawal_notice_component  %>
+<%= render 'components/header-notice',  @content_item.no_deal_notice_component unless @content_item.withdrawal_notice_component %>
```
Comment (alex-ju) [inline app/views/content_items/case_study.html.erb]: nitpick: extra space before `@content_item` (on multiple places in this commit)

File: app/assets/stylesheets/components/_header-notice.scss [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  padding: govuk-spacing(4);
+}
+
+.app-c-header-notice__title {
+  color: govuk-colour("white");
+  margin: 0;
+}
+
+.app-c-header-notice__content {
+  border: 5px solid govuk-colour("black");
```
Comment (alex-ju) [inline app/assets/stylesheets/components/_header-notice.scss]: we can use `$govuk-border-width`

File: app/assets/stylesheets/components/_header-notice.scss [hunk 4]
```diff
@@ -0,0 +1,30 @@
+.app-c-header-notice {
+  margin-bottom: govuk-spacing(8);
+}
+
+.app-c-header-notice__header {
+  background-color: govuk-colour("black");
+  padding: govuk-spacing(4);
+}
+
+.app-c-header-notice__title {
```
Comment (alex-ju) [inline app/assets/stylesheets/components/_header-notice.scss]: we can `@extend %govuk-heading-m;` here and remove the class used in markup

File: app/assets/stylesheets/components/_header-notice.scss [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+.app-c-header-notice__list {
+  margin-bottom: 0;
+}
+
+.app-c-header-notice__link-intro {
+  margin-bottom: govuk-spacing(1);
+}
+
+.app-c-header-notice__link {
```
Comment (alex-ju) [inline app/assets/stylesheets/components/_header-notice.scss]: similarly, we can `@extend %govuk-link;` here and remove the class used in markup

**Pull request merged**